cpp/src/gandiva/decimal_type_util.cc

```diff
@@ -29,8 +29,11 @@ constexpr int32_t DecimalTypeUtil::kMinAdjustedScale;
   }
 
 // Implementation of decimal rules.
+// In addition to having a precision beyond 38, it is compatible with
+// **Redshift's decimal promotion rules** if enable use_redshift_rules.
```

Comment:
Could you use "our compute module's ... rules (which is compatible with the Redshift's ... rules)" or something to be clear?
And could you add the URL that describes the rules?

Reply:
A link to https://arrow.apache.org/docs/cpp/compute.html#arithmetic-functions should be good enough.
Btw, what does "In addition to having a precision beyond 38" mean here?

Reply:
> what does "In addition to having a precision beyond 38" mean here

For decimal128, if the precision beyond 38, compute module's rules will report error with "Decimal precision out of range...".
https://github.com/apache/arrow/blob/54ff758a4570d9eeed9f6195dccdc21dcfe8d6d7/cpp/src/arrow/type.cc#L1471

Gandiva adjust precision to 38 if beyond 38 through `MakeAdjustedType` function, i didn't change the rules here.
Do you think we need to report error like compute module's rules if the precision beyond 38?

Reply:
I think making it consistent with compute is better.

Reply:
Yes, done.